spec/index.adoc

```diff
@@ -626,64 +627,97 @@ See <<state-tree>> for details on the state tree.
+* `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
+If the query call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
 
 * `status` (`text`): `"replied"`
-* `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data `<R>`.
-* `node_signatures` (`[+ node-signature-of-query-response]`): a list of node signatures for the returned query response
-  computed for the following value
-+
-....
-hash_of_map({
-  status: "replied",
-  reply: {arg: <R>},
-  timestamp: <t>,
-  request_id: hash_of_map(content)
-})
-....
-+
-where `hash_of_map` is the <<hash-of-map, representation-independent hash>>, `content` is the CBOR map from the request, and `<t>` is the timestamp of the node when the signature was produced.
+* `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
 * `status` (`text`): `"rejected"`
 * `reject_code` (`nat`): The reject code (see <<reject-codes>>).
 * `reject_message` (`text`): a textual diagnostic message.
 * `error_code` (`text`): an optional implementation-specific textual error code (see <<error-codes>>).
-* `node_signatures` (`[+ node-signature-of-query-response]`): a list of node signatures for the returned query response
-  computed for the following value
-+
+
+[#http-signed-query]
+=== Request: Signed query call
+
+NOTE: The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+A *signed* query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`.
+The request body has the same structure as the request body for <<http-query>> with the `content` map having the following *additional* fields:
+
+* (optional) `return_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"no"`, `"one"`.
+* (optional) `return_certificate` (`text`): Determines whether a certificate (as specified in <<certification>>) is returned in the response; possible values are `"no"`, `"yes"`.
+
+The response is a CBOR (see <<cbor>>) with the same fields as for <<http-query>> response and the following *additional* fields:
+
+* (optional) `signatures` (`[+ node-signature]`): a non-empty list of node signatures for the returned query response.
+* (optional) `certificate` (`blob`): a certificate, as specified in <<certification>>.
+
+Unless `return_signatures` is set to `"no"` in the request, the response to a signed query call contains a non-empty list of signatures
```

Comment:
The list would also be empty if `return_signatures` is omitted, right?

Reply:
fixed